packages/angular-query-experimental/src/query-proxy.ts

```diff
@@ -0,0 +1,50 @@
+import { computed, type Signal } from '@angular/core'
+import type { DefaultError, QueryObserverResult } from '@tanstack/query-core'
+import { CreateBaseQueryResult } from './types'
+
+export function createResultStateSignalProxy<
+  TData = unknown,
+  TError = DefaultError,
+  State = QueryObserverResult<TData, TError>,
+>(resultState: Signal<State>) {
+  const internalState = {} as CreateBaseQueryResult<TData, TError, State>
+
+  return new Proxy(internalState, {
+    get<Key extends keyof State>(
+      target: CreateBaseQueryResult<TData, TError, State>,
+      prop: Key | string | symbol,
+    ) {
+      // first check if we have it in our internal state and return it
+      const computedField = target[prop as Key]
+      if (computedField) return computedField
+
+      // then, check if it's a function on the resultState and return it
+      const targetField = resultState()[prop as Key]
```

Comment:
suggestion: let's `untracked` the `resultState` so we don't unnecessary reinvoking this in some implicit `effect()` 

```suggestion
      const targetField = untracked(resultState)[prop as Key]
```

Reply:
You're right!